app/services/eligibility_service.rb

```diff
@@ -1,90 +1,99 @@
 # frozen_string_literal: true
 
 class EligibilityService
-  def initialize(client, day)
+  def initialize(client, day, number_of_days = 30)
     @client = client
     @day = day
+
+    # How many days to calculate in one Open Fisca request
+    @number_of_days = number_of_days
   end
 
   def run!
     @response = calculate(query)
-  end
 
-  ## Returns 7 days forward from @day, with eligibility as a boolean
-  ## e.g. {'2019-06-01': true, '2019-06-02': true, '2019-06-03': false ... }
-  def meets_minimum_presence_requirements
-    person_result 'citizenship__meets_minimum_presence_requirements'
-  end
+    (0..(@number_of_days - 1)).each do |n|
+      day = (Date.parse(@day) + n).strftime('%Y-%m-%d')
+      person_name = "ruby_#{day}"
 
-  ## Returns 7 days forward from @day, with eligibility as a boolean
-  ## e.g. {'2019-06-01': true, '2019-06-02': true, '2019-06-03': false ... }
-  def meets_each_year_minimum_presence_requirements
-    person_result 'citizenship__meets_each_year_minimum_presence_requirements'
-  end
+      @eligibility = Eligibility.find_or_initialize_by(client: @client, day: day)
 
-  ## Returns 7 days forward from @day, with eligibility as a boolean
-  ## e.g. {'2019-06-01': true, '2019-06-02': true, '2019-06-03': false ... }
-  def meets_5_year_presence_requirement
-    person_result 'citizenship__meets_5_year_presence_requirement'
-  end
+      @eligibility.calculation_data = @response['persons'][person_name]
 
-  ## Returns hash e.g. {'2019-06-01': 365, '2018-06-01': 12, '2017-06-01': 132, '2016-06-01': 12, '2015-06-01': 0 }
-  def days_by_rolling_year
-    person_result 'days_present_in_new_zealand_in_preceeding_year'
-  end
+      # booleans
+      @eligibility.minimum_presence = person_result(person_name, 'citizenship__meets_minimum_presence_requirements')[day]
+      @eligibility.five_year_presence = person_result(person_name, 'citizenship__meets_5_year_presence_requirement')[day]
+      @eligibility.each_year_presence = person_result(person_name, 'citizenship__meets_each_year_minimum_presence_requirements')[day]
+
+      # hash for each rolling year of five
+      @eligibility.present_days_by_rolling_year = person_result person_name, 'days_present_in_new_zealand_in_preceeding_year'
+      @eligibility.mimimum_presence_by_rolling_year = person_result person_name, 'citizenship__meets_preceeding_single_year_minimum_presence_requirement'
+
+      @eligibility.save!
+
+      Rails.logger.info @eligibility
+    end
 
-  def enough_days_by_rolling_year
-    person_result 'citizenship__meets_preceeding_single_year_minimum_presence_requirement'
+    @eligibility
   end
 
-  private
+  def days_to_iterate; end
 
-  def person_result(key)
+  def person_result(person_name, key)
     @response['persons'][person_name][key]
   end
 
+  private
+
   def query
+    all_persons = {}
+    all_names = []
+    (0..@number_of_days).each do |n|
+      day = (Date.parse(@day) + n).strftime('%Y-%m-%d')
```

Comment:
This is a repeat from line 16, it should probably be pulled out into its own method.